docs/specs/configuration.md

```diff
@@ -53,6 +59,16 @@ under entities and also used to define global Vector behavior.
 
 ### Naming
 
+#### Flag naming
+
+- MUST only contain ASCII alphanumeric, lowercase, and hyphens
+- MUST be in kebab-case format when multiple words are used (e.g., `config-dir`)
+- For flags that take a value, but are also able to be "disabled", they SHOULD NOT use a sentinel
+  value. Instead they SHOULD have a second flag added prefixed with `no-` and SHOULD leave off any
+  unit suffixes. For example, to disable `--graceful-shutdown-limit-secs`,
```

Comment:
The configuration spec suggests the unit suffixes should be _complete_ (seconds over secs), while that's not really followed in our configuration today - is that something we'd want unified between config and CLI?

Reply:
Good point. I think `_secs` is more common than `_seconds` at this point. We should reconfirm for future options. I'll do that outside of this PR.

Reply:
It definitely is - it's probably worth considering just changing the spec to be shortened units rather than changing _every_ configuration 😂 

Outside of this PR is 👍 just figured worth raising given the smol overlap.